### PR TITLE
Actions

### LINKED ISSUE #4: Automate Npm Audit Fix
Schedule regular, automated fixes (and post-fix testing) to the extent possible, perhaps using GitHub Actions.

### LINKED ISSUE #24: Set up Actions for test automation

### LINKED ISSUE #25: Publish code documentation to gitpages
Blocker - need to get GitHub Actions up and running, then use existing npm script job to publish code documentation.

### PR DESCRIPTION
**Pull request by lanaebk**
* Set up event-triggered workflow to build, lint, publish code docs, and test on push or PR to master 
* Set up scheduled workflow to run npm audit once a week
  - https://github.com/marketplace/actions/npm-audit-action
  - if any issues are found, this action will open an issue titled "npm audit found vulnerabilities" with the `vulnerability` label

Resolves #24, resolves #4

### CODE REVIEW

**Comment by yonkeltron** [comment]
Hey looks pretty cool. Love how all the checks are passing here.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    
+    - name: Use Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    
+    - name: Cache node modules
```
Comment (yonkeltron) [inline .github/workflows/main.yml:22]: I sort of love how clear this is.

Comment (lanaebk) [reply .github/workflows/main.yml:22]: Right?!

**Review by yonkeltron — APPROVED** [review]
Approved.

**Pull request merged**